vhost-device-sound/src/audio_backends/pipewire/test_utils.rs

```diff
@@ -132,3 +135,60 @@ fn print_output(child: &mut Child, id: &'static str) -> Option<()> {
 
     None
 }
+
+pub fn truncated_wait_delay<D: rand::distributions::Distribution<f32>, R: rand::Rng>(
+    slot_time: &Duration,
+    attempts_so_far: u32,
+    exponent_max: u32,
+    rng: &mut R,
+    distribution: &D,
+) -> Duration {
+    #[allow(clippy::cast_possible_wrap)]
+    let attempts_so_far: i32 = attempts_so_far.clamp(0_u32, exponent_max) as i32;
+    let position = distribution.sample(rng);
+    let max = 2_f32.powi(attempts_so_far) - 1.0_f32;
+    slot_time.mul_f32(position * max)
+}
+
+pub fn try_backoff<T, E: std::fmt::Display>(closure: impl Fn() -> Result<T, E>) -> T {
+    const NO_DELAY: Duration = Duration::new(0, 0);
+
+    let mut iterations: u32 = 0;
+    let mut dur: Option<Duration> = None;
+    let mut rng = rand::thread_rng();
+
+    let distribution = Uniform::new(0.0_f32, 1.0_f32);
+
+    loop {
```

Comment:
What about putting a limit on the number of iterations?
time limit or count limit

Reply:
Can do, already have that change in my git stash ;)

Reply:
@epilys apart from this, the rest LGTM.

Reply:
Added!